src/megaapi_impl.cpp

```diff
@@ -23127,6 +23669,116 @@ void MegaSyncPrivate::setRegExp(MegaRegExp *regExp)
     }
 }
 
+int MegaSyncPrivate::getError() const
+{
+    return mError;
+}
+
+void MegaSyncPrivate::setError(int error)
+{
+    mError = error;
+}
+
+void MegaSyncPrivate::disable(int error)
+{
+    setState(SYNC_DISABLED);
+    setError(error);
+}
+
+bool MegaSyncPrivate::isEnabled() const
+{
+    return state != SYNC_CANCELED && (state != SYNC_DISABLED || mError != NO_SYNC_ERROR );
```

Comment:
:thinking: Will a sync considered Enabled when it has not been canceled, but disabled with a temporary error?

Reply:
yes. as the header says, isEnabled means is enabled bby the user. 

```
    /**
     * @brief Returns if the sync is set as enabled by the user
     *
     * Notice that this will return true even when the sync is failed (fatal error)
     * or temporary disabled (transient error/circumstance).
     *
     * @return if the sync is set as enabled by the user
     */
    virtual bool isEnabled() const;```